internal/controllers/v1beta1/gateway.go

```diff
@@ -50,12 +93,147 @@ func (c *GatewayController) SetupWithManager(ctx context.Context, mgr manager.Ma
 	return nil
 }
 
-func (r *GatewayController) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+func (c *GatewayController) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
 	// set up a convenient log object so we don't have to type request over and over again
 	log := log.FromContext(ctx)
+	log.Info("Reconciling Gateway...", "reconcile", request.String())
+	log.V(1).Info("Debug")
+	gateway, err := c.istioClient.NetworkingV1beta1().Gateways(request.Namespace).Get(ctx, request.Name, metav1.GetOptions{})
 
-	log.Info("Reconciling Gateway...")
-	// TODO not implemented
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil // garbage collection will handle
+		}
+
+		log.Error(err, "Error reconciling gateway, requeued")
+		return reconcile.Result{
+			Requeue: true,
+		}, err
+	}
+
+	for _, s := range gateway.Spec.Servers {
+		log.V(1).Info("Inspecting server", "hosts", s.Hosts)
+
```

Comment:
Should the check for if `server.Tls.Mode = v1beta1.ServerTLSSettings_SIMPLE` then skip on L153 be made here instead since this loop has both the Create and Update?

Is it possible we update a certificate that isn't `server.Tls.Mode != v1beta1.ServerTLSSettings_SIMPLE`?

Reply:
I viewed this condition as a concern for the action of "Creating a Certificate", not a reconcile loop. 

A certificate is only updated if it exists.